test/PackageVersionsTest/E2EInstallerTest.php

```diff
@@ -187,6 +187,51 @@ public function testRemovingPluginWithNoDevDoesNotAttemptToGenerateVersions() :
         );
     }
 
+    public function testInstallingPluginWithNoScriptsOverridesOriginalRequirements() : void
+    {
+        $this->createPackageVersionsArtifact();
+
+        $this->writeComposerJsonFile(
+            [
+                'name'         => 'package-versions/e2e-transitive',
+                'require'      => [
+                    'ocramius/package-versions' => '1.0.0',
+                    'infection/infection' => '^0.15.0',
+                ],
+                'repositories' => [
+                    ['packagist' => false],
+                    [
+                        'type' => 'artifact',
+                        'url' => $this->tempArtifact,
+                    ],
+                    [
+                        'type' => 'vcs',
+                        'url' => 'https://github.com/scaytrase/package-versions-main-stub.git',
```

Comment:
Is there a way to include the package definitions manually? This test shouldn't rely on network access

Reply:
Sure, this is just a dirty implementation of the problem. I think I can to replace it with `path` or `artifact` generated in test
https://getcomposer.org/doc/05-repositories.md#path

Reply:
Please, check. I'm generating local zip artifacts now, refactored some code for it into reusable function

Reply:
I can do further refactorings, if you need them (i.e. refactor out logic on generating and validating  package version asserts)